src/Compiler/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -538,7 +539,18 @@ public static IDisposable BuildLinePragma(this CodeWriter writer, SourceSpan? sp
             return NullDisposable.Default;
         }
 
-        return new LinePragmaWriter(writer, span.Value, context, 0, false);
+        var spanValue = span.Value;
+        if (context.Options.RemapDesignTimeLinePragmaPathsOnWindows && RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
Should we do the same for enhanced line pragmas below?

Reply:
Oh, we probably don't since they are not used in design time.

Reply:
Nothing stops future developers using enhanced line pragmas during design time either. In fact I think there is a tooling bug with rename that is fixed by doing so.